controllers/topolvm_csi_driver.go

```diff
@@ -0,0 +1,99 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	driverName = "csi-driver"
+)
+
+type csiDriver struct{}
+
+// csiDriver unit satisfies resourceManager interface
+var _ resourceManager = csiDriver{}
+
+func (c csiDriver) getName() string {
+	return driverName
+}
+
+func (c csiDriver) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	csiDriverDeployment := getCSIDriverDeployment()
+	result, err := cutil.CreateOrUpdate(ctx, r.Client, csiDriverDeployment, func() error {
+		return nil
+	})
+	// CSIDriver resource is an immutable resource and can have result either Created or Unchanged and it's clusterscoped
+	switch result {
+	case cutil.OperationResultCreated:
+		r.Log.Info("csi driver", "operation", result, "name", csiDriverDeployment.Name)
+	case cutil.OperationResultNone:
+		r.Log.Info("csi driver", "operation", result, "name", csiDriverDeployment.Name)
+	default:
+		r.Log.Error(err, "csi driver reconcile failure", "name", csiDriverDeployment.Name)
+		return err
+	}
+	return nil
+}
+
+func (c csiDriver) ensureDeleted(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	csiDriverDeployment := &storagev1.CSIDriver{}
```

Comment:
This is not a deployment. Please rename to csiDriver.

Reply:
changing it to csiDriverResource to avoid name clashes.

Reply:
changing it to `csiDriverResource` to avoid name clashes.